pkg/kv/kvserver/protectedts/update.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package protectedts
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/kv"
+	"github.com/cockroachdb/cockroach/pkg/util/hlc"
+)
+
+// UpdateFn is a callback that updates a protected timestamp record.
+type UpdateFn func(txn *kv.Txn, r RecordMetadata, ru *RecordUpdater) error
```

Comment:
Do we really need all of this machinery? Could we just have an `UpdateTimestamp` method?

Reply:
Might be overkill, do you see a future where we would update spans too? Even then maybe it is simpler to just have methods for each field we want to update.

Reply:
Perhaps we will, but I wonder if, in that world, we just want to delete and then create with the same ID.

Reply:
Changed to just having a UpdateTimestamp method.